core/src/subsystems/flywheel.cpp

```diff
@@ -0,0 +1,71 @@
+/*********************************************************
+*
+*     File:     Flywheel.cpp
+*     Purpose:  Generalized flywheel class for Core.
+*     Author:   Chris Nokes
+*     
+**********************************************************
+* EDIT HISTORY
+**********************************************************
+* 09/16/2022  <CRN> Created file, added constructor, spins, RPM setting, stop.
+* 09/18/2022  <CRN> Added async functionality.
+*********************************************************/
+
+#include "../core/include/subsystems/flywheel.h"
+#include "../core/include/utils/pid.h"
+#include "vex.h"
+#include <thread>
+
+using namespace vex;
+
+/*
+* Create the Flywheel object.
+*/
+Flywheel::Flywheel(motor_group &motors, robot_specs_t &config, PID &pid)
+         :motors(motors), config(config), pid(pid){ }
+
+// Spin motors using voltage; defaults forward at 12 volts
+// Speed is between -1 and 1.
+void Flywheel::spin(double speed, directionType dir){
+  motors.spin(dir, speed * 12, voltageUnits::volt);
+}
+
+// spin this flywheel at a given RPM, async; runs until stop(), stopThread(), or a new spinRPM() is called. 
+int spinRPMThread(void* wheelPointer) {
+  Flywheel* wheel = (Flywheel*) wheelPointer;
+  wheel->getPID()->set_target((double) wheel->getRPM());  // get the pid from the wheel and set its target to the RPM stored in the wheel.
+  while(true) {
+    wheel->getPID()->update(wheel->getMotors()->velocity(velocityUnits::rpm));        // check the current velocity and update the PID with it.
+    wheel->getMotors()->spin(fwd, wheel->getPID()->get() * 12, voltageUnits::volt);   // set the motors to whatever PID tells them to do
+    vexDelay(20);
+  }
+  return 0; // only here to make the compiler SHUT UP
+}
+
+// starts / restarts RPM thread at new value
+void Flywheel::spinRPM(int inputRPM) {
```

Comment:
If someone puts this in the main loop, it'll be starting/stopping the thread over and over again, which could cause issues. Consider following something like [the task management in Lift.h](https://github.com/RIT-VEX-U/Core/blob/9a9c7c9806ded60ed169a6e8e3d4fc26bc2eeae8/include/subsystems/lift.h#L63). It starts the task in the constructor and only changes the target in set() methods.

You can also replace the getPID and getMotors with a single get_target method in this case